docs/architecture/adr-050-sign-mode-textual-annex1.md

```diff
@@ -0,0 +1,275 @@
+- Allow optional annotation for textual name (TBD)
+- Algorithm:
+  - convert enum name (`VoteOption`) to snake_case (`VOTE_OPTION`)
+  - truncate that prefix + `_` from the enum name if it exists (`VOTE_OPTION_` gets stripped from `VOTE_OPTION_YES` -> `YES`)
+  - convert rest to sentence case: `YES` -> `Yes`
+  - in summary: `VOTE_OPTION_YES` -> `Yes`
+
+#### Examples
+
+See example above with `message Vote{}`.
+
+### `google.protobuf.Any`
+
+- Applies to `google.protobuf.Any`
+- Rendered as:
+
+```
+<type_url>
+> <value rendered underlying message>
+```
+
+#### Examples
+
+TODO
+
+### `google.protobuf.Timestamp`
+
+Rendered as either ISO8601 (`2021-01-01T12:00:00Z`).
+
+### `google.protobuf.Duration` (TODO)
+
+- rendered in terms of weeks, days, hours, minutes and seconds as these time units can be measured independently of any calendar and duration values are in seconds (so months and years can't be used precisely)
+- total seconds values included at the end so users have both pieces of information
+- Ex:
+  - `1483530 seconds` -> `2 weeks, 3 days, 4 hours, 5 minutes, 30 seconds (1483530 seconds total)`
+
+### bytes
+
+- Bytes are rendered in base64.
+
+### address bytes
+
+We currently use `string` types in protobuf for addresses so this may not be needed, but if any address bytes are used in sign mode textual they should be rendered with bech32 formatting
+
+### strings
+
+Strings are rendered as-is.
+
+### Default Values
+
+- Default Protobuf values for each field are skipped.
+
+#### Example
+
+```proto
+message TestData {
+  string signer = 1;
+  string metadata = 2;
+}
+```
+
+```go
+myTestData := TestData{
+  Signer: "cosmos1abc"
+}
+```
+
+We get the following encoding for the `TestData` message:
+
+```
+TestData object
+> Signer: cosmos1abc
+```
+
+### [ABANDONED] Custom `msg_title` instead of Msg `type_url`
+
+_This paragraph is in the Annex for informational purposes only, and will be removed in a next update of the ADR._
+
+<details>
+  <summary>Click to see abandoned idea.</summary>
+
+- all protobuf messages to be used with `SIGN_MODE_TEXTUAL` CAN have a short title associated with them that can be used in format strings whenever the type URL is explicitly referenced via the `cosmos.msg.v1.textual.msg_title` Protobuf message option.
+- if this option is not specified for a Msg, then the Protobuf fully qualified name will be used.
+
+```proto
+message MsgSend {
+  option (cosmos.msg.v1.textual.msg_title) = "bank send coins";
+}
+```
+
+- they MUST be unique per message, per chain
+
+#### Examples
+
+- `cosmos.bank.v1beta1.MsgSend` -> `bank send coins` (no v1beta1 in string, for bijectivity we map this string to v1beta1)
+- `cosmos.bank.v1.MsgSend` -> `bank v1 send coins` (if `cosmos.bank.v1` is released and used in parallel with `cosmos.bank.v1beta1`)
+- `cosmos.gov.v1beta1.MsgVote` -> `governance vote` (no v1beta1 in string, for bijectivity we map this string to v1beta1)
+- `cosmos.gov.v1.MsgVote` -> `governance v1 vote`
```

Comment:
I think we can remove this paragraph or make it shorter - not need to provide examples for thing which is abandoned.